Da-Bois-Tracking/Tracking-App
# TA-16 Incorporate FastAPI

### In this PR, I'm adding/changing the following:

- Incorporating FastAPI to use as our backend.

### Please do the following to test my code:

1. Checkout to my feature branch:
   `git checkout TA-16-Incorporate-FastAPI`

2. Change directory to the Backend folder:
   `cd backend`

3. Start your Virtual Environment
   - For Windows:
      `python -m venv venv`
      `venv/Scripts/activate`
   - For macOS / Linux:
      `python3 -m venv venv`
      `source venv/bin/activate`

4. Install Dependencies:
   `pip install -r requirements.txt`
   If prompted, make sure to upgrade your pip.

5. Start the Application:
   `uvicorn main:app --reload`

6. Verify backend changes
   - Once your server starts, open up a web page and navigate to `localhost:8000`.
      From here, you should see a JSON object with:
   ```
      {
         "Hello": "World"
      }
   ```
   
   - Navigate to `localhost:8000/docs`
      From here, you should see a FastAPI page with a "Greetings" tag and a `/Hello` endpoint.
![image](https://github.com/Da-Bois-Tracking/Tracking-App/assets/57964857/b3d5dcb5-8959-4697-a403-2fb9c389d2fc)

   - Click on the `/hello` endpoint to open up its dropdown.
   - Click on "Try it out"
   - Click on Execute

 After executing the endpoint, you will be met with a 200 code and a response body with the object of
```
{
   "Hello": "World"
}
```
![image](https://github.com/Da-Bois-Tracking/Tracking-App/assets/57964857/3affc88a-6867-49e7-b6a6-0024a533d92c)


### Review Checklist
- [x] Switch to the feature branch.
- [x] Follow the installation instructions to set up the project and its dependencies.
- [x] Verify the FastAPI application runs successfully and is accessible.
- [x] Check `localhost:800` and `localhost:8000/docs` to confirm the expected output.

[Related Jira Ticket](https://bois-tracking-app.atlassian.net/browse/TA-16)
